### PR TITLE
Add testing setup + server explanation to readme

### PR DESCRIPTION
**Pull request by boxgames1**
- Add testing setup with RTL. What we use, what most candidates actually use as well. Let them spend time on the important part of the tech test, not setting up
- Smoke test
- Modified the README to say that we prefer the filtering to be done in the server.

Open to suggestions and changes

### CODE REVIEW

File: README-FRONTEND.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ For this challenge we would like you to:
 
 - Populate items in the left-hand sidebar with data from the /api/items endpoint.
 - Allow users to select items in the left-hand sidebar and have them appear in the Menu Preview on the right hand side.
-- Allow users to filter items in the left-hand sidebar by name.
+- Allow users to filter items in the left-hand sidebar by name. We prefer this to be done on the server.
```
Comment (mdsomerfield) [inline README-FRONTEND.md]: I would change this to [TIP: you can modify the server code]

File: src/client/App.test.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import { render, screen } from '@testing-library/react'
+import App from './App'
+
+// You can remove this test
```
Comment (mdsomerfield) [inline src/client/App.test.js]: I think you could remove the comment and just call the test 'Example test' and that should be implied.

File: README-FRONTEND.md (hunk 1 above)
Comment (boxgames1) [reply README-FRONTEND.md]:
```suggestion
- Allow users to filter items in the left-hand sidebar by name. You can modify the server code.
```

File: src/client/App.test.js (hunk 2 above)
Comment (boxgames1) [reply src/client/App.test.js]:
```suggestion
// Example test
```

**Review by mdsomerfield — APPROVED** [review]

**Pull request merged**